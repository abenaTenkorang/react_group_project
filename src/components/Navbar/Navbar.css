* {
  box-sizing: border-box;
}

body {
  margin-top: 30px;
}

.nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 2rem;
  padding: 0 3rem;
  background-color: white;
  height: 50px;
}

.nav ul {
  padding: 0;
  margin: 0;
  list-style: none;
  display: flex;
  justify-content: center;
  gap: 0.3rem;
}

.nav a {
  color: blue;
}

.title {
  display: flex;
  flex-direction: row;
  justify-content: center;
  column-gap: 10px;
  color: black;
  border: none;
  text-decoration-line: none;
}

.logo {
  margin-top: 20px;
  height: 50px;
}

.title > h1 {
  font-size: 30px;
  font-weight: 900;
}
