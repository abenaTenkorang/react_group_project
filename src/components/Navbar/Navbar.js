import { Link, useMatch, useResolvedPath } from 'react-router-dom';
import React from 'react';
import PropTypes from 'prop-types';
import './Navbar.css';

const Navbar = () => (
  <nav className="nav">
    <div className="title">
      <img src="./planet.png" alt="SpaceTravellers' Hub" className="logo" />
      <h1>SpaceTravellers Hub</h1>
    </div>
    <ul>
      <CustomLink to="/Rockets">Rockets</CustomLink>
      <CustomLink to="/Missions">Missions</CustomLink>
      <CustomLink to="/MyProfile">MyProfile</CustomLink>
    </ul>
  </nav>
);

function CustomLink({ to, children, className = '' }) {
  const resolvedPath = useResolvedPath(to);
  const isActive = useMatch({ path: resolvedPath.pathname, end: true });
  return (
    <li>
      <Link to={to} className={`${className} ${isActive ? 'active' : ''}`}>
        {children}
      </Link>
    </li>
  );
}

CustomLink.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

CustomLink.defaultProps = {
  className: '',
};

export default Navbar;
