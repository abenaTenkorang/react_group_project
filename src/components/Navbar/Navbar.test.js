import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from './Navbar';
import '@testing-library/jest-dom';

describe('Navbar component', () => {
  test('renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('navigates to the correct page on click', () => {
    const { getByText } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    fireEvent.click(getByText('Rockets'));
    expect(window.location.pathname).toBe('/Rockets');
  });

  test("applies 'active' class to the active link", () => {
    const { getByText } = render(
      <BrowserRouter initialEntries={['/Rockets']}>
        <Navbar />
      </BrowserRouter>,
    );
    const rocketsLink = getByText('Rockets');
    const missionsLink = getByText('Missions');
    const myProfileLink = getByText('MyProfile');

    expect(rocketsLink).toHaveClass('active');
    expect(missionsLink).not.toHaveClass('active');
    expect(myProfileLink).not.toHaveClass('active');

    fireEvent.click(missionsLink);

    expect(rocketsLink).not.toHaveClass('active');
    expect(missionsLink).toHaveClass('active');
    expect(myProfileLink).not.toHaveClass('active');
  });
});
