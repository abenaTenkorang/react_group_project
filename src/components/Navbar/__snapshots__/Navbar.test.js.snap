// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly 1`] = `
<nav
  class="nav"
>
  <div
    class="title"
  >
    <img
      alt="SpaceTravellers' Hub"
      class="logo"
      src="./planet.png"
    />
    <h1>
      SpaceTravellers Hub
    </h1>
  </div>
  <ul>
    <li>
      <a
        class=" "
        href="/Rockets"
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        class=" "
        href="/Missions"
      >
        Missions
      </a>
    </li>
    <li>
      <a
        class=" "
        href="/MyProfile"
      >
        MyProfile
      </a>
    </li>
  </ul>
</nav>
`;
